pb-crm-api/src/app.module.ts

```diff
@@ -2,38 +2,48 @@ import {Module} from '@nestjs/common';
 import {AuthnModule} from './authn/authn.module';
 import {TypeOrmModule} from "@nestjs/typeorm";
 import {ContactModule} from "./contact/contact.module";
-import { UsersModule } from './authn/users/users.module';
-import {ConfigModule} from "@nestjs/config";
-import { UploadController } from './upload/upload.controller';
-import { UploadService } from './upload/upload.service';
-import { UploadModule } from './upload/upload.module';
-import { MulterModule } from '@nestjs/platform-express';
-
+import {UsersModule} from './authn/users/users.module';
+import {ConfigModule, ConfigService} from "@nestjs/config";
+import {UploadModule} from './upload/upload.module';
+import {MulterModule} from '@nestjs/platform-express';
 import {ErrorService} from "./core/services/error.service";
 
+
+
 @Module({
     imports: [
-        ConfigModule.forRoot(),
         MulterModule.registerAsync({
-          useFactory: () => ({
-            dest: './files'
-          }),
+            useFactory: () => ({
+                dest: './files'
+            }),
+        }),
+        TypeOrmModule.forRootAsync({
+            imports: [ConfigModule.forRoot({
+                envFilePath: `./config/.env.${process.env.ENV}`
+            })],
+            useFactory: (configService: ConfigService) => ({
+                type: 'postgres',
+                host: configService.get('DB_HOST'),
+                port: +configService.get<number>('DB_PORT'),
+                username: configService.get('DB_USR'),
+                password: configService.get('DB_PWD'),
+                database: configService.get('DB_NAME'),
+                entities: ["dist/db/entities/**/*.entity{.js, .ts}"],
+                synchronize: true,
+                dropSchema: configService.get<boolean>('DB_DROP'),
+                logging: true,
+                migrations: ["../dist/db/migrations/*{.js,.ts}"],
+                seeds: ["../dist/db/seeders/**/*.js"],
+                factories: ["../dist/db/factories/**/*.js"],
+                cli: {
+                    migrationsDir: "src/db/migrations"
+                },
+                ssl: process.env.ENV === 'dev' ? {
```

Comment:
Any reason why you did this variable this way and used config service for the others?

Reply:
Basically, the local setup won't run if any ssl is enabled/defined, so even if it was true or false it would still fail. I tried to set it up to run a config that would establish it then pass the entire thing through the app module, but it wasn't working correctly... so this just basically enables it only for dev, and says undefined otherwise..

Reply:
👍